test/test_atomic.cu

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * @file test_atomic.cu
+ * @brief Unit test for atomic operations.
+ * @author Andrei Gheata (andrei.gheata@cern.ch)
+ */
+
+#include <iostream>
+#include <cassert>
+#include <AdePT/Atomic.h>
+#include <VecGeom/backend/cuda/Interface.h>
+
+// Example data structure containing several atomics
+struct SomeStruct {
+  adept::Atomic_t<int> var_int{nullptr};
+  adept::Atomic_t<float> var_float{nullptr};
```

Comment:
It is quite unintuitive to have `adept::Atomic_t<float>` actually be a pointer type.

Reply:
This is a leftover to be fixed